minor corrections and HTC10 device added

pintaf opened the pull request.

utzcoz commented on README.md:
```diff
@@ -8,7 +8,7 @@ with Git](https://source.android.com/source/version-control.html).
 To initialize your local repository using the Maru OS trees, use a command like this:
 
-    repo init -u git://github.com/maruos/manifest.git -b maru-0.6
+    repo init -u git://github.com/maruos/manifest.git -b maru-0.7
```
A new PR for this modification has been merged. Maybe you should rebase to the newest code base.

utzcoz commented on default.xml:
```diff
@@ -26,15 +26,24 @@
   <project path="packages/apps/LineageParts" name="android_packages_apps_LineageParts" remote="maruos" />
   <project path="prebuilts/busybox" name="platform_prebuilts_busybox" remote="maruos" revision="master" />
-
+  <project name="LineageOS/android_packages_resources_devicesettings" path="packages/resources/devicesettings" remote="github" revision="lineage-16.0" />
```
Nit: could you move it to the top of maruos part?

pintaf replied:
I could but that's not really a maruos thing. You would hit the same problem and require this repo while building classic LOS.

Where do you suggest to put it ?
above lxc or above vendor/maruos ?

utzcoz replied:
Maybe.

pdsouza replied:
Hmmm, it's not a huge deal of course but it would be nice if `default.xml` was purely Maru repos. 

Here are some options:

1. The only other file we have where we specify customizations is in `snippets/overrides.xml`. At the bottom of that file we remove some unnecessary LOS repos for example. So it's possible it could go in there but I guess this isn't really an "override".
2. We could possibly create another file in the `snippets` directory called `extras.xml` or something and add it in there.
3. We can just move it to either above or below all the `maruos` repos in `default.xml`.

Any opinions which would be best?

utzcoz replied:
I prefer to choice 3, because we can only maintain one file for manifest. And the next is choice 2, we can include it with `include` tag.

pdsouza replied:
I prefer choice 3 as well. @pintaf, can you move this above all the `maruos` repos in our manifest?

pintaf replied:
I also think choice 3 is the best. However instead of above, I think bellow the maruos repos would be better.
Indeed, this repo need came with my HTC10, but it is also needed by some other devices. Hence it is mostly device related and should be closer between maru repos and devices repos in my opinion.

I'll integrate my HTC repos in maruos and update this PR. Maybe tomorrow, maybe Thursday, I'm sinking under the things to do.

pdsouza replied:
OK below is fine with me especially since it is device-related. Take your time on the PR, no rush.

utzcoz commented on default.xml:
```diff
@@ -26,15 +26,24 @@
+  <!-- END hammerhead - nexus 5 -->
+
+  <!-- START pme - HTC 10 -->
+  <project path="device/htc/pme" name="android_device_htc_pme" remote="pintaf" />
```
Maybe you should push those projects to maruos. Hi @pdsouza , could you help to grant permission to @pintaf ?

pdsouza replied:
Yes, I will add @pintaf to our team.

pintaf committed:
added HTC 10 repos

Signed-off-by: Loïc Poisot <elpoisot@hotmail.fr>

pintaf committed:
added a repo to solve missing string errors for some devices:

device/htc/pme/devicesettings/res/values/arrays.xml:20: error: resource string/action_none (aka org.lineageos.settings.device:string/action_none) not found.
device/htc/pme/devicesettings/res/values/arrays.xml:21: error: resource string/action_launch_camera (aka org.lineageos.settings.device:string/action_launch_camera) not found.
device/htc/pme/devicesettings/res/values/arrays.xml:22: error: resource string/action_torch (aka org.lineageos.settings.device:string/action_torch) not found.
device/htc/pme/devicesettings/res/values/styles.xml:38: error: resource layout/preference_category_material_settings (aka org.lineageos.settings.device:layout/preference_category_material_settings) not found.
device/htc/pme/devicesettings/res/values/styles.xml:49: error: resource layout/preference_material_settings (aka org.lineageos.settings.device:layout/preference_material_settings) not found.
error: failed linking references.

Signed-off-by: Loïc Poisot <elpoisot@hotmail.fr>

pintaf commented:
@utzcoz PR updated, the HTC10 repo will be updated later once I get access to maruos, or you can integrate them yourself based on my repos.

pdsouza commented:
Hi @pintaf, I've just invited you to join our org so you can create your HTC10 repos. Let me know if you have any issues.

pintaf committed:
added Google Pixel phones and updated HTC10 repos to maruos

Signed-off-by: Loïc Poisot <elpoisot@hotmail.fr>

pintaf commented:
Updated, added Pixels repo as well

pdsouza commented on default.xml:
```diff
@@ -27,14 +27,29 @@
+    <!-- START Sailfish/Marlin - Google Pixel / Google Pixel XL -->
+  <project path="device/google/marlin" name="android_device_google_marlin" remote="maruos" />
+  <project path="kernel/google/marlin" name="android_kernel_google_marlin" remote="maruos" />
+  <project path="vendor/google" name="proprietary_vendor_google" remote="maruos" revision="lineage-16.0" />
```
We don't usually include the vendor blob repos in our manifest and usually let devs select whichever method they want to use of obtaining these (including stripping them from factory images with either android-generate-vendor or android-prepare-vendor).

But having the vendor blobs forked in and set up by default in our manifest may make our community's development efforts easier since devices just build out of the box. I don't mind trying this approach for maru-0.7.

Any thoughts, @utzcoz?

pintaf replied:
if I may, advanced devs know how to get blobs from other sources if they need to.
Including them eases things for people just getting started in building/porting as it was my case.

Maru clearly needs support and committed people. Making things as easy as possible is a way to attract / keep devs.

utzcoz replied:
Vendor proprietary blob maybe incur `DMCA` from GitHub. What about we update the porting document with the method to get and add those files to specific devices, and add document to our Q&A part to devs?

pintaf replied:
If DMCA is a problem and I understand it can be, maybe we should not host the blobs in the maru org, but still include links to some proprietary repo so that devices can build out of the box. For example https://github.com/TheMuppets/
This way if some company has to complain against it's blobs being hosted they will ask github to close TheMuppets repo.
So no risks, but clearly improved developer experience.

pdsouza replied:
I can see arguments both ways. Yes, the dev experience is easier since devices will build out of the box, but on the downside devs will not know the process for obtaining vendor blobs themselves, which is important knowledge if you want to know what's actually happening under the hood.

My main issue with including links to those specific vendor repos in our official manifest may be seen as endorsing them, which we do not do in any way. They are useful to get going quickly, but the files really should be pulled from a trusted factory image or running device.

Given that LineageOS itself does not ship these blob files in their official manifest, I think it's reasonable to assume that Maru OS devs should learn how to obtain these on their own. As @utzcoz suggests, we can update our developer docs with very simple and clear instructions on how to get these blobs, and in the device-specific guide (e.g. [Pixel](https://maruos.com/docs/devices/sailfish.html#install-maru-via-twrp)) we can give explicit instructions on how to obtain the blobs for that device. Also, for the Google devices like the Pixel, we usually use [android-prepare-vendor](https://github.com/anestisb/android-prepare-vendor) to grab these blobs from the official factory image, which is a trusted source.

Sorry @pintaf, but let's take the conservative approach for now. Let's remove the forked blob repos from our org, and remove the manifest links. I'm happy to merge everything else. We can discuss this further later too, but let's at least merge the rest of the PR while we figure out how best to deal with this.

pintaf replied:
Ok @pdsouza I removed the vendor blobs. Ill try to delete the associated repos from maru org-
Would be nice if you can merge that I have some time to work on this tonight.

pdsouza reviewed (commented):
Overall looks good to me. Just waiting on @utzcoz's approval before merging.

utzcoz commented:
> Overall looks good to me. Just waiting on @utzcoz's approval before merging.

This weekend I will test @pintaf 's awesome work for porting work for sailfish and marlin with my devices.

pintaf commented:
> > Overall looks good to me. Just waiting on @utzcoz's approval before merging.
> 
> This weekend I will test @pintaf 's awesome work for porting work for sailfish and marlin with my devices.

I don't have the devices with me, so it was a "blind" port. I hope it wil boot

pintaf committed:
removed vendor repos

Signed-off-by: Loïc Poisot <elpoisot@hotmail.fr>

pdsouza reviewed (approved):
LGTM. Thanks @pintaf for the quick update!
